src/strategies/priority_strategy.rs

```diff
@@ -616,24 +577,143 @@ impl UniswapXPriorityFill {
                 }
             }
         }
+
+        // After processing orders, check if any can be submitted
+        return self.check_orders_for_submission().await
     }
 
-    async fn try_send_order_batch(
+    async fn try_route_order_batch(
         &self,
         order_batch: OrderBatchData,
         order_hash: String,
-        order_data: OrderData,
     ) {
         match self.batch_sender.send(vec![order_batch]).await {
             Ok(_) => (),
             Err(e) => {
                 error!(
-                    "{} - Failed to send batch: {}; moving order back to new_orders",
+                    "{} - Failed to send batch: {}",
                     order_hash, e
                 );
-                self.processing_orders.remove(&order_hash);
-                self.new_orders.insert(order_hash.clone(), order_data);
             }
         }
     }
+
+    async fn check_orders_for_submission(&self) -> Vec<Action> {
+        let order_hashes: Vec<String> = self.new_orders
+            .iter()
+            .map(|entry| entry.key().clone())
+            .collect();
+
+        let mut actions = Vec::new();
+
+        for order_hash in order_hashes {
+            if let Some(order_data) = self.new_orders.get(&order_hash) {
+                // Skip if no route available
+                if order_data.route.is_none() {
+                    continue;
+                }
+                // skip if order is already in processing_orders
+                if self.processing_orders.contains_key(&order_hash) {
+                    continue;
+                }
+
+                // Check if order is now fillable
+                let order = match &order_data.order {
+                    Order::PriorityOrder(order) => order,
+                    _ => continue,
```

Comment:
does this mean we'll skip to the next iteration in the loop 🤨 ? kinda weird syntax but neat